improve docs for akka-http #20699

note opened the pull request:
Made imports in docs snippets more complete and a few minor improvements

Related to https://github.com/akka/akka/issues/20699

note committed:
improve docs for akka-http #20699

Made imports in docs snippets more complete and a few minor improvements

ktoso commented:
OK TO TEST

ktoso commented on akka-docs/rst/scala/http/introduction.rst:
```diff
@@ -25,9 +25,8 @@ Akka HTTP was designed specifically as “not-a-framework”, not because we don
+Akka HTTP is provided in a separate jar file, to use it make sure to include the following dependency::
 
-  "com.typesafe.akka" %% "akka-http-core" % "@version@" @crossString@
   "com.typesafe.akka" %% "akka-http-experimental" % "@version@" @crossString@
```
Not quite sure about this change... We do want to explain that it's 2 modules.

How about explaining here that: hey, you should likely use `"com.typesafe.akka" %% "akka-http-experimental" % "@version@" @crossString@` however you should know that the core is actually in a separate module?

note replied:
See your point. I added some explanation in the following paragrapth. Does it state clearly what you meant?

ktoso replied:
Yes, very nicely worded - thanks!

ktoso commented:
LGTM besides https://github.com/akka/akka/pull/20722/files#r65899632
Let's address that and merge.

Thanks a lot @note !

note commented:
I addressed your comment. Also I changed sth in `WebSocketClientExampleSpec.scala` and `WebSocketClientExampleTest.java` because I saw some complaints on gitter.

ktoso commented on akka-docs/rst/scala/code/docs/http/scaladsl/WebSocketClientExampleSpec.scala:
```diff
@@ -23,59 +20,60 @@ class WebSocketClientExampleSpec extends WordSpec with Matchers with CompileOnly
   }
 
   "half-closed-WebSocket-closing-example" in compileOnlySpec {
+    import akka.actor.ActorSystem
```
thanks :) that one's from the chat today I think :)

note replied:
Exactly

ktoso commented:
LGTM, let's merge if validation is happy. 
Thanks for the PR @note !

ktoso commented:
Thanks!
